aries_vcx/src/handlers/proof_presentation/verifier/state_machine.rs

```diff
@@ -54,12 +54,14 @@ impl VerifierSM {
         }
     }
 
-    pub fn from_request(source_id: &str, presentation_request: &PresentationRequestData) -> Self {
-        Self {
+    // todo: eliminate VcxResult (follow set_request err chain and eliminate possibiltiy of err at the bottom)
```

Comment:
Why? `set_request` may always be called in invalid state

Reply:
The chain of errors origins in 
```
impl Json {
    pub fn new(id: AttachmentId, json: serde_json::Value, encoding: AttachmentEncoding) -> VcxResult<Json> {
         ....
    }
```
because we refuse to base64 encode anything else but `serde_json::Value::Object` and `serde_json::Value::String`. Why don't we for instance allow base64 encoding of JSON arrays? 

I've removed this restriction and therefore also chain of errors. But if you want us to restrict encoded types to lets say JSON Object, JSON Array and String, we can still do that without possibility of returning error. The code simplified nicely after removing `VcxResult<>` wrap everywhere

Reply:
oh I damn ofc `set_request` has this
```
_ => { return Err(VcxError::from_msg(VcxErrorKind::InvalidState, "Cannot set presentation request in this state")) }
```
as the second source of error...nothing that can be done with that.